hledger-lib/Hledger/Data/Dates.hs

```diff
@@ -257,7 +266,7 @@ earliest (Just d1) (Just d2) = Just $ min d1 d2
 -- | Parse a period expression to an Interval and overall DateSpan using
 -- the provided reference date, or return a parse error.
 parsePeriodExpr :: Day -> Text -> Either (ParseError Char MPErr) (Interval, DateSpan)
-parsePeriodExpr refdate = parsewith (periodexpr refdate <* eof)
+parsePeriodExpr refdate s = parsewith (periodexpr refdate <* eof) (lc s)
```

Comment:
For `stack8.0.yaml` (my distro still use `8.0.2`):
```
    /home/nikolay/contrib/hledger/hledger-lib/Hledger/Data/Dates.hs:269:68: error:
        • Couldn't match type ‘[Char]’ with ‘Text’
          Expected type: Text
            Actual type: String
        • In the second argument of ‘parsewith’, namely ‘(lc s)’
          In the expression: parsewith (periodexpr refdate <* eof) (lc s)
          In an equation for ‘parsePeriodExpr’:
              parsePeriodExpr refdate s
                = parsewith (periodexpr refdate <* eof) (lc s)
```

Reply:
Should be fixed now